### ISSUE TITLE
truffle migrate got stuck after generating transaction hash

### ISSUE THREAD

**Issue by ChristobelDiana** [ChristobelDiana/truffle#1]
I tried to deploy metacoin project. After unlocking my account and start mining,I called truffle migrate, it got stuck after displaying transaction hash. I used geth rpc.
Kindly help solving this issue.
![image](https://user-images.githubusercontent.com/46812189/56456209-b5e59400-6386-11e9-9449-0bd56c6b7ca4.png)

**Comment by rob-Hitchens** [comment]
Can you please expand the view. We need more information about the underlying processes here. Let's see the miner?

To give an example of what we might be looking for, net42 uses a very low initial difficulty. Given Ethereum's target block time of 14 seconds, eventually difficulty with self-adjust and the blocks will stop coming (over compensation). A work around to that is to `init`, mine, `truffle migrate --reset`. I'm not sure that is the problem or the solution. We just can't tell from looking at this. 

If you are using terminator, you can get multiple windows open - miner, truffle, more ... alternatively, please arrange things so we can see what you see. Every indication is mining is not happening so let's see why we think it is.

**Comment by ChristobelDiana** [comment]
/home/christobel/Pictures/Screenshot from 2019-04-21 15-09-23.png

**Comment by ChristobelDiana** [comment]
![Screenshot from 2019-04-21 15-20-46](https://user-images.githubusercontent.com/46812189/56468385-3a005000-6449-11e9-89b0-5ae9f363c439.png)

**Comment by ChristobelDiana** [comment]
> Can you please expand the view. We need more information about the underlying processes here. Let's see the miner?
> 
> To give an example of what we might be looking for, net42 uses a very low initial difficulty. Given Ethereum's target block time of 14 seconds, eventually difficulty with self-adjust and the blocks will stop coming (over compensation). A work around to that is to `init`, mine, `truffle migrate --reset`. I'm not sure that is the problem or the solution. We just can't tell from looking at this.
> 
> If you are using terminator, you can get multiple windows open - miner, truffle, more ... alternatively, please arrange things so we can see what you see. Every indication is mining is not happening so let's see why we think it is.


![](https://user-images.githubusercontent.com/46812189/56468331-99119500-6448-11e9-91bc-507cb8eab1cf.png)

![](https://user-images.githubusercontent.com/46812189/56468385-3a005000-6449-11e9-89b0-5ae9f363c439.png)

**Comment by ChristobelDiana** [comment]
Ok. hereby I have attached the screenshots. I unlocked my account, started
mining and tried truffle migrate --reset too. but again I get the same
problem.

**Comment by rob-Hitchens** [comment]
Please reinitialized the net42 with `--init` and test again. I have a suspicion and this is the fastest method to resolve it.

**Comment by CallMeGwei** [comment]
Can we see a copy of your net42 config? I remember that I had an error in mine that caused some non-obvious issues...

**Comment by ChristobelDiana** [comment]
geth --datadir ~/.ethereum/net42 init ~/DAPPS/genesis42.json
geth --datadir ~/.ethereum/net42 --networkid 42 --rpc --rpcport 8545
--rpcaddr 0.0.0.0 --rpccorsdomain "*" --rpcapi "eth,net,web3" --gcmode
archive console

**Comment by CallMeGwei** [comment]
Yeah, that looks like exactly what I have in my launcher script... so nothing wrong there.

From your screenshots, it seems that at least one contract is being sent to geth and is, in fact, being deployed despite the truffle message. Are you able to jump in to the `truffle console` and pull up that transaction hash manually?

Upon closer inspection... in your whole desktop screenshot, window top left... what is all of that _above_ `1_initial_migrations.js` ?

For comparison... when I `truffle migrate --reset` I see this:

![image](https://user-images.githubusercontent.com/34364155/56854696-5811f880-6900-11e9-8fa9-b8da80bb8cfe.png)

The "Summary" line comes after the migrate command has finished. What's weird about your screenshot, to me, is that it's showing "summary" and then continues to process *another* migrate run without accepting any user input in between.

It might be nothing, but it is one of the only "out of the ordinary" things I can see from what you have provided so far.

**Comment by ChristobelDiana** [comment]
Actually, that summary line came for migrations dry -run(simulation). Did you get this simulation when migrating?
Thanks anyway. I have tried configuring with different network id in windows and it works now.

**Comment by CallMeGwei** [comment]
Never got any simulation while running migrations.

Glad you got it sorted!

**Comment by xavierlepretre** [comment]
Is the issue still open?

**Comment by ChristobelDiana** [comment]
> Is the issue still open?

The issue is not sorted out. Could not find the reason why this error occurred.Found alternative solution in windows

**Comment by xavierlepretre** [comment]
I don't see any mining being done. No block is being formed. I suspect that the net42 genesis block was not put there.
You need to `geth --datadir /where/ever removedb` then launch again.
Look carefully at the log, it will tell you if it put the proper genesis file or if it defaulted to the main net one.

**Comment by stackchain** [comment]
Facing the same issue here.
